docs: align Rust guide structure with JS (#4)
rticommunity/rticonnextdds-connector-rust: docs/guide/connector.md

> # Connector lifecycle

Reviewer: I don't understand use of the term "lifecycle" here. Could the intro describe what the Connector lifecycle is/the purpose of this chapter?

> [`crate::Connector`] represents a DDS `DomainParticipant` configured from XML.

Reviewer: ```suggestion
To import the crate:
[`crate::Connector`] represents a DDS `DomainParticipant` configured from XML.
```

It owns native resources and creates `Input` and `Output` handles.

## Importing the crate

```rust

> use rtiddsconnector::Connector;

Reviewer 2: This would be `use rtiddsconnector;` but that's not needed. If you want to import things, you'd use `use rtiddsconnector::{self, Connector};` to import the crate and some element from the crate.

```

## Creating a connector

Create a connector with [`crate::Connector::new`], passing the namespaced
participant name and the XML file path:

```rust
use rtiddsconnector::Connector;

fn create_connector() -> rtiddsconnector::ConnectorFallible {
    let _connector = Connector::new("MyLibrary::MyParticipant", "App.xml")?;
    Ok(())
}
```

The XML file defines your types, QoS profiles, and DDS entities. The call above

> loads a `<domain_participant>` from a `<domain_participant_library>`, for example:

Reviewer: ```suggestion
loads a `<domain_participant>` from a `<domain_participant_library>`. For example:
```

```xml
<domain_participant_library name="MyParticipantLibrary">
  <domain_participant name="MyParticipant" domain_ref="MyDomainLibrary::MyDomain">
    ...
  </domain_participant>
</domain_participant_library>
```

See a complete example in `examples/MyApplication.xml`.

## Closing a connector

There is no explicit `close()` method in Rust. Instead, `Connector`, `Input`, and
`Output` are released when they go out of scope. The crate uses RAII to free
native resources.

> If you want to force cleanup of Connext global resources at the end of a scope

Reviewer: ```suggestion
To force cleanup of Connext global resources at the end of a scope
```

> (for example in tests), use [`crate::GlobalsDropGuard`].

Reviewer: ```suggestion
(for example, in tests), use [`crate::GlobalsDropGuard`].
```

## Getting inputs and outputs

[`crate::Connector::get_input`] and [`crate::Connector::get_output`] return
owned handles to the named reader/writer. The crate enforces single-threaded
ownership: if an entity is already in use, these calls return an error. To wait
until the entity becomes available, use [`crate::Connector::take_input`] or
[`crate::Connector::take_output`].

See [Publishing data](crate::guide::output) and [Reading data](crate::guide::input)
for the workflow that follows.

> ## Threading note

Reviewer: If appropriate in this context, consider formatting this as a note instead of a separate heading.

> Operations on the same `Connector` or its contained entities are not protected

Reviewer: ```suggestion
Multiple operations on the same `Connector`, or its contained entities, are not protected
```

for multi-threaded access. See [Threading and ownership](crate::guide::threading)
for guidance.

> ## Auto-enable considerations

Reviewer: Are these considerations applicable to all entities? If not:

```suggestion
## DataReader auto-enable considerations
```

Connext DDS controls entity enablement through XML QoS settings (for example,
`<entity_factory>/<autoenable_created_entities>`). If readers are enabled before
you acquire an `Input`, data may already be available when you call

> `get_input`. This behavior comes from Connext configuration rather than the

Reviewer: ```suggestion
`get_input`. This behavior stems from your Connext configuration rather than the
```
